src/org/zalando/stups/pierone/api_v2.clj

```diff
@@ -176,7 +186,7 @@
                   (-> (ring/response "")
                       (ring/status 201)))
                 (do
-                  (resp "image not found" request :status 404))))))
+                  (resp {:errors [(assoc-in (:BLOB_UNKNOWN errors) [:detail] {"Digest" digest})]} request :status 404))))))
```

Comment:
Could you wrap the generation of this object in a function? Something along the lines of

~~~ clojure
(defn get-error-response [error-id & args]
  (condp = error-id
    :BLOB_UNKOWN (apply generate-blob-unknown-error args) ; or do it here directly
    :MANIFEST_UNKNOWN (apply generate-manifest-unknown-error args)))
~~~